Marlin/Marlin_main.cpp

```diff
@@ -5922,18 +5929,17 @@ inline void gcode_M17() {
   }
 
   static void ensure_safe_temperature() {
-    bool did_show = false;
     wait_for_heatup = true;
     while (wait_for_heatup) {
       idle();
+      if (!wait_for_heatup) break;
```

Comment:
I may be mistaken, but I thought the call to `idle()` was to allow the emergency parser to execute. If that's the case, then if the emergency parser sees a M108, then that M108 will set `wait_for_heatup` to **false**, which would immediately be overwritten by the line right after the call to `idle() `without this change.